Kernel/Interrupts/APIC.cpp

```diff
@@ -223,9 +243,58 @@ void enable(u32 cpu)
 
             IO::delay(200);
         }
+
+        // Now wait until the ap_cpu_init_pending variable dropped to 0, which means all APs are initialized and no longer need these special mappings
+        if (m_apic_ap_count.load(AK::MemoryOrder::memory_order_consume) != aps_to_enable) {
+            klog() << "APIC: Waiting for " << aps_to_enable << " AP(s) to finish initialization...";
+            do {
+                // Wait a little bit
+                IO::delay(200);
```

Comment:
:(

Reply:
?

Reply:
Could we maybe not delay boot-up while the application CPUs are starting?

Perhaps this could be done in a separate thread or something.

Reply:
200 microseconds is not too bad. Doesn't seem worth complicating this by bringing in the scheduler while booting AP's.

Reply:
Notice this is in a loop :(